docs/semgrep-code/remove-duplicates.md

```diff
@@ -5,26 +5,116 @@ title: Remove duplicate findings
+Semgrep generates a finding whenever it scans a project, and one of its rules matches a piece of code in the project. Since Semgrep scans a project multiple times, it needs a way to track the same finding in a file over time. Semgrep does this using two types of fingerprints: `syntactic_id` and `match_based_id`.
 
-If a finding is fixed in one branch (such as `main`) but open in another (such as `production`), and the code fixes are present in both branches, initiate a scan through your CI job or SCM tool on the branch(es) with open findings to have Semgrep mark the findings as fixed.
+
+:::info
+The calculations used to determine whether findings are new are subject to change at any time as Semgrep improves its deduplication logic.
+:::
+
+### `syntactic_id`
+
+Semgrep generates the `syntactic_id` for a finding using the following information for a finding:
+
+- The file path
+- The name of the rule that generated the finding
+- The code syntax, or the literal piece of code that matched the rule
+- The index, a value generated by determining the number of times the rule involved matched code in the file
+
+This information is combined and then hashed before storing for privacy.
+
+Using the `syntactic_id`, Semgrep can determine if a given finding in a file is 
+the same as a finding identified during a different scan, even if the code snippet that the rule matched had been moved to a different location in the file. This allows Semgrep to avoid generating a new finding and deduplicating its records accordingly, even across multiple branches associated with the project. It also means that Semgrep can cross-correlate findings, so a finding that has been triaged in one branch will be flagged as triaged if it's identified in another branch.
+
+### `match_based_id`
+
+While the `syntactic_id` associated with a finding helps track findings within the same file, there are times when the way Semgrep generates the `syntactic_id` isn't sufficient for the identification and deduplication of findings. 
+
+Semgrep generates the `match_based_id` for a finding using the following information for a finding:
+
+- The file path
+- The name of the rule that generated the finding
+- The rule pattern with the metavariables' values substituted in
+
+This information is combined and then hashed. At this point, Semgrep appends the **index**, a value generated by determining the number of times the rule involved matched code in the file. Note that the index is appended to the hash, not combined with the other finding information before hashing. This is done to preserve information on how findings are related. For example, `finding0` with `match_based_id = 123_0` and `finding1` with `match_based_id = 123_0` indicate that both were generated from the same rule matching the same code in the same file.
```

Comment:
This is confusing. There should rarely even be two distinct findings with the same match-based ID. If you want to explain the index, a better approach would be to change one of the indices to 1, and explain that these are the same rule matching two instances of similar code (remember, match-based ID doesn't use the literal code, it uses the rule pattern) in the same file.

Reply:
Sorry, the identical index was a typo!